src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -173,13 +179,109 @@ public String getDisplayName() {
       return "Mesos Cloud";
     }
 
-    // TODO: validate URLs
+    /**
+     * Validates that the Mesos master URL is a valid URL.
+     *
+     * @param mesosMasterUrl The Mesos master URL supplied by the user.
+     * @return Whether the URL is valid or not.
+     */
+    public FormValidation doCheckMesosMasterUrl(@QueryParameter String mesosMasterUrl) {
+      // This will change with https://jira.mesosphere.com/browse/DCOS-53671.
+      if (isValidUrl(mesosMasterUrl)) {
+        return FormValidation.ok();
+      } else {
+        return FormValidation.error(mesosMasterUrl + " is not a valid URL.");
+      }
+    }
+
+    /**
+     * Validates that the framework name is not empty.
+     *
+     * @param frameworkName The framework name set by the user.
+     * @return Whether the framework name is empty or not.
+     */
+    public FormValidation doCheckFrameworkName(@QueryParameter String frameworkName) {
```

Comment:
we chose to strip spaces then instead of validate their absence? (I guess for backwards compat we need to do the former?)

Reply:
I'll add space stripping.